src/sentry/integrations/github/client.py

```diff
@@ -102,21 +103,38 @@ def get_tree(self, repo_full_name: str, tree_sha: str) -> JSONData:
 
         return tree
 
-    def get_trees_for_org(self, org_name: str) -> JSONData:
+    def get_trees_for_org(self, gh_org: str, cache_seconds: int = 3600 * 24) -> JSONData:
         """
         This fetches tree representations of all repos for an org.
         """
         trees: JSONData = {}
-        repositories = self.get_repositories(fetch_max_pages=True)
-        # XXX: In order to speed up this function we will need to parallelize this
-        # Use ThreadPoolExecutor; see src/sentry/utils/snuba.py#L358
-        for repo_info in repositories:
-            full_name: str = repo_info["full_name"]
-            branch = repo_info["default_branch"]
-            trees[full_name] = {"default_branch": branch, "files": self.get_tree(full_name, branch)}
+        cache_key = f"githubtrees:repositories:{gh_org}"
+        repo_key = "githubtrees:repo"
+        # XXX: We do not support more than one org atm
+        cached_repositories = cache.get(cache_key, [])
+        if not cached_repositories:
+            repositories = self.get_repositories(fetch_max_pages=True)
+            # XXX: In order to speed up this function we will need to parallelize this
+            # Use ThreadPoolExecutor; see src/sentry/utils/snuba.py#L358
+            for repo_info in repositories:
+                full_name: str = repo_info["full_name"]
+                cached_repositories.append(full_name)
+                branch = repo_info["default_branch"]
+                trees[full_name] = {
+                    "default_branch": branch,
+                    "files": self.get_tree(full_name, branch),
+                }
+                cache.set(f"{repo_key}:{full_name}", trees[full_name], cache_seconds)
+            cache.set(cache_key, repositories, cache_seconds)
+            logger.info(f"Caching trees for {gh_org} org until TBD.")
```

Comment:
Reminder to replace the TBD. I think something like this would work https://github.com/getsentry/sentry/blob/04d79898820ca6c5c9bbf3cd5a8bf57ced0c88e0/src/sentry/tasks/post_process.py#L621